src/neo/Ledger/Blockchain.cs

```diff
@@ -302,6 +366,10 @@ private void Persist(Block block)
         {
             using (SnapshotCache snapshot = GetSnapshot())
             {
+                if (block.Index != 0 && block.Index == HeaderHeight + 1)
+                {
+                    recentHeaders.Add(block.Header);
```

Comment:
The header should be removed from the cache after that it has been persisted so that we can accept new headers.

Reply:
fixed